File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    cnt = 1
    task_1 = test_cnt(progresses[0], speeds[0])
    
    for i in range(1, len(progresses)):
        task_2 = test_cnt(progresses[i], speeds[i])
        if task_1 < task_2:
            answer.append(cnt)
            cnt = 1
            task_1 = task_2
        else:
            cnt += 1
    answer.append(cnt)
    return answer

def test_cnt(p, s):
    cnt = 0
    while p < 100:
        p += s
        cnt += 1
    
    return cnt

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int max_time = (100-progresses[0] + speeds[0] - 1) / speeds[0];
    int cnt = 1;
    for(int i = 1; i < progresses.size(); i++) {
        if ((100-progresses[i] + speeds[i] - 1) / speeds[i] > max_time) {
            max_time = (100-progresses[i] + speeds[i] - 1) / speeds[i];
            answer.push_back(cnt);
            cnt = 1;
        }
        else{
            cnt+=1;
        }
    }
    answer.push_back(cnt);
    return answer;
}